test/creator.cpp

```diff
@@ -101,34 +101,67 @@ TEST(ZimCreator, createEmptyZim)
   header.read(*reader);
   ASSERT_FALSE(header.hasMainPage());
 #if defined(ENABLE_XAPIAN)
-  ASSERT_EQ(header.getArticleCount(), 1); // xapiantitleIndex
-
-  //Read the only one item existing.
-  auto urlPtrPos = offset_t(header.getUrlPtrPos());
-  auto direntOffset = offset_t(reader->read_uint<offset_type>(urlPtrPos));
+  entry_index_type nb_entry = 3; // xapiantitleIndex and titleListIndexes (*2)
+  int xapian_mimetype = 0;
+  int listing_mimetype = 1;
+#else
+  entry_index_type nb_entry = 2; // titleListIndexes (*2)
+  int listing_mimetype = 0;
+#endif
+  ASSERT_EQ(header.getArticleCount(), nb_entry);
 
   DirentReader direntReader(reader);
-  auto dirent = direntReader.readDirent(direntOffset);
-  test_article_dirent(dirent, 'X', "title/xapian", "title/xapian", 0, cluster_index_t(0), blob_index_t(0));
-#else
-  ASSERT_EQ(header.getArticleCount(), 0);
+  auto urlPtrPos = header.getUrlPtrPos();
+  offset_t direntOffset;
+  std::shared_ptr<const Dirent> dirent;
+
+  direntOffset = offset_t(reader->read_uint<offset_type>(offset_t(urlPtrPos)));
+  dirent = direntReader.readDirent(direntOffset);
```

Comment:
I would introduce a helper class `DirentTableReader` that combines `DirentReader` with `urlPtrPos` (which denotes the beginning of a contiguous dirent pointer table). Then these two lines (and their multiple duplications below) can be replaced by much more concise `direntReader.readDirent(direntIndex)`.

Reply:
Indeed, and it is in fact the purpose of `DirectDirentAccessor`. Let's use it in the tests.